src/ica_api.c

```diff
@@ -363,7 +363,7 @@ unsigned int ica_open_adapter(ica_adapter_handle_t *adapter_handle)
 	}
 	if (*adapter_handle != -1) {
 		/* Test if character device is accessible. */
-		if (!ioctl(*adapter_handle, Z90STAT_STATUS_MASK, &status_mask)) {
+		if (!ioctl(*adapter_handle, ZCRYPT_STATUS_MASK, &status_mask)) {
 			return 0;
 		}
 	}
```

Comment:
What about this instead:

diff --git a/src/ica_api.c b/src/ica_api.c
index c9ea1ad..74e25b9 100644
--- a/src/ica_api.c
+++ b/src/ica_api.c
@@ -345,7 +345,8 @@ static unsigned int check_message_part(unsigned int message_part)
 
 unsigned int ica_open_adapter(ica_adapter_handle_t *adapter_handle)
 {
-       char *name, status_mask[64];
+       struct zcrypt_device_matrix_ext status_mask;
+       char *name;
 
        if (!adapter_handle)
                return EINVAL;
@@ -363,7 +364,7 @@ unsigned int ica_open_adapter(ica_adapter_handle_t *adapter_handle)
        }
        if (*adapter_handle != -1) {
                /* Test if character device is accessible. */
-               if (!ioctl(*adapter_handle, Z90STAT_STATUS_MASK, &status_mask)) {
+               if (!ioctl(*adapter_handle, ZCRYPT_DEVICE_STATUS, &status_mask)) {
                        return 0;
                }
        }

Reply:
Or you use your code but replace the status_mask[256] with status_mask[MAX_ZDEV_CARDIDS_EXT].
But I see that the ZCRYPT_STATUS_MASK ioctl will become deprecated soon also.